packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -301,15 +301,15 @@ export class ScheduleManager {
                 return;
             }
 
-            // If only length one then clear
+            // If the batch contains only a single op, clear the batch flag.
             if (messages.length === 1) {
-                delete messages[0].metadata;
+                delete firstMessageMetadata.batch;
```

Comment:
First bug: should delete the 'batch: true' flag, not the entire message metadata.